client/src/+assets/assets-list.tsx

```diff
@@ -42,15 +44,67 @@ export function AssetsList({ assets }: AssetsListProps) {
     setBatchSelected(batchSelected.concat(...dids.filter(did => !batchSelected.includes(did))))
   }
 
+  const checkError = (message: string) => {
+    if(message.includes('"statusCode":401')) {
+      newLogin(sdk, loginMarketplaceAPI)
+      setErrorMessage('Your login is expired. Please first sign with your wallet and after try again')
+    } else {
+      setErrorMessage(message)
+    }
+    
+    popupRef.current?.open()
+  }
+
+  const onAddBookmark = async (did: string, description: string) => {
+    try {
+      const bookmark = await sdk.bookmarks.create({
+        did,
+        userId: userProfile.userId,
+        description,
+      });
+  
+      setBookmarks([...bookmarks, bookmark])
+    } catch (error: any) {
+      checkError(error.message)
+    }
+  }
+
+  const onRemoveBookmark = async (did: string) => {
+    try {
+      const bookmark = bookmarks.find(item => item.did === did );
+
+      if (bookmark?.id) {
+        await sdk.bookmarks.deleteOneById(bookmark?.id);
+        setBookmarks(bookmarks.filter(item => item.id !== bookmark?.id))
```

Comment:
I don't think we should use `?` inside the if block

Reply:
It was `prettier` plugin that included it, I will removed